api/lib/infrastructure/repositories/certification-assessment-repository.js

```diff
@@ -14,13 +14,12 @@ async function _getCertificationChallenges(certificationCourseId, knexConn) {
     .orderBy('challengeId', 'asc');
 
   const challengeIds = certificationChallengeRows.map(({ challengeId }) => challengeId);
-  const allChallenges = await challengeRepository.getMany(challengeIds);
+  const challengesType = await challengeRepository.getManyTypes(challengeIds);
 
   return _.map(certificationChallengeRows, (certificationChallengeRow) => {
-    const challenge = _.find(allChallenges, { id: certificationChallengeRow.challengeId });
     return new CertificationChallengeWithType({
       ...certificationChallengeRow,
-      type: challenge?.type,
+      type: challengesType[certificationChallengeRow.challengeId],
```

Comment:
On a ici un mini  changement de comportement potentiel, le code précédent prenait en compte le fait qu'on pouvait ne pas trouver le challenge. Je ne  connais pas de cas ou cela arriverait, mais je le signale.

Reply:
Je ne pense pas que le comportement change vraiment, car si le challenge n'est pas trouvée, `type` devrait être `undefined` comme avant.

Reply:
Bien vu.

Reply:
En paracourant l'historique du fichier il est pas clair pour moi la raison de l'introduction du `'EmptyType'`.

Sur Metabase je vois :

```
select count(1) from "challenges"
where "type" is NULL;

-- 50

select count(1) from "certification-challenges"
inner join "challenges" on "challenges"."id" = "certification-challenges"."challengeId"
where "challenges"."type" IS NULL;

-- Cette requête est impossible à faire via MEtabase (Timeout metabase), du coup si quelqu'un sait comment avoir le résultat
```

Si la deuxième requête venait à retourner un résultat, pour moi ça rendrait dangereux le changement proposé. Si elle ne retourne rien, alors en théorie oui pas de cas existant, par contre on s'expose quand même au soucis en jour peut-être.

La question qui pourrait aider à trancher : pourquoi avons-nous actuellement en PROD des types à NULL en base ?


PS : même ça Metabase arrive pas à executer 😭 `select DISTINCT "challengeId" from "certification-challenges" where EXISTS ( select DISTINCT "id" from "challenges" where "challenges"."type" IS NULL and "challengeId" = "challenges"."id" LIMIT 1 ) LIMIT 1;`

Reply:
C'est une recherche intéressante.
Pour en revenir au sujet initial, le cas qui nous intéresse c'est surtout pourquoi nous ne trouverions nous pas une épreuve.

Reply:
On a regardé pour les 50 challenges avec un type non défini. Ils sont tout en status proposé ou périmé. Non jouable actuellement donc.